all-in-one-apim/modules/integration-v2/docs/devs/lint_features.py

```
@@ -0,0 +1,327 @@
+    hard = [x for x in violations if x[2] in HARD]
+    advisory = [x for x in violations if x[2] not in HARD]
+    for (path, line, code, msg) in hard:
+        print(f"{os.path.relpath(path, MODULE)}:{line}: [{code}] {msg}")
+    if show_all:
+        for (path, line, code, msg) in advisory:
+            print(f"{os.path.relpath(path, MODULE)}:{line}: [{code}][advisory] {msg}")
+    hby = Counter(c for (_, _, c, _) in hard)
+    aby = Counter(c for (_, _, c, _) in advisory)
+    print(f"\n[feature-lint] {len(hard)} hard violation(s) across {nfiles} feature files"
+          + (": " + ", ".join(f"{k}={v}" for k, v in sorted(hby.items())) if hard else " (clean)")
+          + (f"  |  advisory (non-gating): " + ", ".join(f"{k}={v}" for k, v in sorted(aby.items()))
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)